src/lib_json/json_writer.cpp

```diff
@@ -409,7 +409,7 @@ void FastWriter::writeValue(const Value& value) {
   case objectValue: {
     Value::Members members(value.getMemberNames());
     document_ += '{';
-    for (Value::Members::iterator it = members.begin(); it != members.end();
+    for (auto it = members.begin(); it != members.end();
```

Comment:
Another nice way to do it avoids exposing the iterators
or special-casing the first element.

```
const char* sep = "";
const char* const kvSep = yamlCompatibilityEnabled_ ? ": " : ":";
for (const auto& name : members) {
  document += sep;
  sep = ",";
  document_ += valueToQuotedStringN(name.data(),
                                    static_cast<unsigned>(name.length()));
  document_ += kvSep;
  writeValue(value[name]);
}

```

Reply:
I'll let you make this change as a separate PR.  The requested changes are an improvement beyond the scope of simple use of auto.  I agree that your solution is 2 steps better, but I want to take 1 step at a time.